darwin/exporter/formats/yolo_segmented.py

```diff
@@ -78,14 +79,25 @@ class YoloSegmentedAnnotationType(Enum):
     POLYGON = auto()
 
 
-def _determine_annotation_type(data: dict, annotation_index: int) -> YoloSegmentedAnnotationType:
-    if "x" in data and "y" in data and "w" in data and "h" in data:
-        return YoloSegmentedAnnotationType.BOUNDING_BOX
-    elif "points" in data:
-        if isinstance(data["points"][0], list):
-            logger.warn(f"Skipped annotation at index {annotation_index} because it's a complex polygon'")
-            return YoloSegmentedAnnotationType.UNKNOWN
+def _determine_annotation_type(annotation: Annotation) -> YoloSegmentedAnnotationType:
+    """
+    Determines the annotation type
+
+    Parameters
+    ----------
+    annotation : Annotation
+        The annotation to be determined.
 
+    Returns
+    -------
+    YoloSegmentedAnnotationType
+        The annotation type.
+    """
+    type = annotation.annotation_class.annotation_type
+
+    if type == "bounding_box":
+        return YoloSegmentedAnnotationType.BOUNDING_BOX
+    elif type == "polygon":
```

Comment:
is complex_polygon/circle/other annotation types not applicable to YOLO?

Reply:
Largely speaking, Yolo has historically been bounding boxes, and now supports polygons too, but ellipses, points, and other types are not supported - with many (2D plane) types, we could still process the bounding box and include it, but this wasn't in scope of the feature, and it was decided to include only bbox and simple polys for now.